tests/route/test_route_perf.py

```diff
@@ -91,8 +123,8 @@ def exec_routes(duthost, prefixes, str_intf_nexthop, op):
     generate_route_file(duthost, prefixes, str_intf_nexthop, route_file_dir, op)
 
     # Check the number of routes in ASIC_DB
-    start_num_route = int(duthost.shell('sonic-db-cli ASIC_DB eval "return #redis.call(\'keys\', \'{}*\')" 0'.format(ROUTE_TABLE_NAME))['stdout'])
-    
+    start_num_route = count_routes(duthost)
+
     # Calculate timeout as a function of the number of routes
     route_timeout = max(len(prefixes) / 500, 1) # Allow at least 1 second even when there is a limited number of routes
```

Comment:
Thanks for this great work! I did see a device that fails to finish the routes within this time limit and does not have enough resources for an additional 10000 routes. I thought this might not be an image issue. And we probably want to relax the timeout a bit and reduce the default number of routes.

Reply:
I also noticed error reported by swssconfig when applying a large config file.
```
Oct 26 05:40:54.180200 str-dx010-acs-4 NOTICE swss#swssconfig: :- main: Loading config from JSON file:/arp_test.json...
Oct 26 05:41:00.540328 str-dx010-acs-4 ERR swss#swssconfig: :- main: Exception caught: Unable to connect to redis: Cannot assign requested address
```
Someone has reported the issue here https://github.com/Azure/sonic-buildimage/issues/3849 .